Automattic/jetpack
# [Boost]: Add settings system to Boost Cache.

This PR adds a settings system to Boost Cache which is shared between the auto loaded Page_Cache Module and the caching system loaded by advanced-cache.php.

Fixes [#35216](https://github.com/Automattic/jetpack/issues/35216)
## Proposed changes:

* This adds the Boost_Class_Settings class which includes the get() and set() methods.
* It will have an internal $settings arrray.
* get() will return the value of one settings at a time.
* set() accepts an associative array of settings to save to the config file. The internal $settings array is updated at the same time.
* The config file is stored in wp-content/boost-cache/ and is named config.php
* When run for the first time, it will create a new config file with the setting `enabled => true`.
* The config file is a php script and the first line is <?php die(); The second line is simply a JSON encoded array of the settings.
* The config file is loaded using the file() command.

### Other information:


## Jetpack product discussion
pc9hqz-2rA-p2

## Does this pull request change what data or activity we track or use?
No

## Testing instructions:
* Apply PR
* Reload your blog.
* Check for the new config file at wp-content/boost-cache/config.php
